### PR TITLE
OCM-2489: Fix useSpotInstances attribute usage

### PR DESCRIPTION
**Pull request by nirarg**
* Refactor the code, to make it more clear
* Populate state should set useSpotInstances only if it true, otherwise leave it empty
* Prevent setting negative "max_spot_price"

### CODE REVIEW

File: provider/machine_pool_resource.go [hunk 1]
```diff
@@ -455,30 +455,29 @@ func (r *MachinePoolResource) Update(ctx context.Context, request tfsdk.UpdateRe
 	response.Diagnostics.Append(diags...)
 }
 
-func getSpotInstances(state *MachinePoolState, mpBuilder *cmv1.MachinePoolBuilder) (
-	useSpotInstances bool, errMsg string) {
-	useSpotInstances = false
+func setSpotInstances(state *MachinePoolState, mpBuilder *cmv1.MachinePoolBuilder) error {
+	useSpotInstances := !state.UseSpotInstances.Unknown && !state.UseSpotInstances.Null && state.UseSpotInstances.Value
```
Comment (bardielle) [inline provider/machine_pool_resource.go:459]:
```suggestion
	useSpotInstances := !helper.IsStringAttributeEmpty(state.UseSpotInstances)
```

File: provider/machine_pool_resource.go [hunk 2]
```diff
@@ -455,30 +455,29 @@ func (r *MachinePoolResource) Update(ctx context.Context, request tfsdk.UpdateRe
...
+	isSpotMaxPriceSet := !state.MaxSpotPrice.Unknown && !state.MaxSpotPrice.Null
```
Comment (bardielle) [inline provider/machine_pool_resource.go:460]:
```suggestion
	isSpotMaxPriceSet := !helper.IsStringAttributeEmpty(state.MaxSpotPrice)
```

File: provider/machine_pool_resource.go (hunk 1 above)
Comment (nirarg) [reply provider/machine_pool_resource.go:459]: but useSpotInstances is boolean, not string

**Review by bardielle — APPROVED** [review]
/lgtm

**Pull request merged**